swap_meet/vendor.py

```diff
@@ -1,2 +1,125 @@
+        return item
+    
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+
+    def get_by_id(self,id):
+        """
+        Wave 02: Provides method to retrieve an item by its ID.
+        Method: get_by_id(id) searches inventory for matching item ID.
+        Returns the item if found, otherwise returns None.
+        """
+        for item in self.inventory:
+            if item.id == id:
+                return item       
+        return None
+
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        """
+        Wave 03: Handles swapping items between two vendors.
+        Method: swap_items(other_vendor, my_item, their_item) exchanges items between inventories.
+        Returns False if either item is not found; otherwise completes the swap.
+        """
+
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        self.remove(my_item)
+        other_vendor.add(my_item)
+        other_vendor.remove(their_item)
+        self.add(their_item)
+        return True
+    
+    
+    def swap_first_item(self, other_vendor):
+        """
+        Wave 04: Swaps the first item between two vendors.
+        Method: swap_first_item(other_vendor) exchanges the first item of each inventory.
+        Returns True if swap is successful; returns False if either inventory is empty.
+        """
+        
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]
+        return True
+
+    
+    def get_by_category(self, category):
+        """
+        Wave 06: Provides category-based item management and swapping between vendors.
+        Methods: get_by_category(category) returns all items in the category; 
+        get_best_by_category(category) returns the item with highest condition or None; 
+        swap_best_by_category(other_vendor, my_priority, their_priority) swaps best-matching items and returns True or None if no match.
+        """
+        
+        category_list = []
+        
+        if not category:
+            return category_list
+
+        for item in self.inventory:
+            if item.get_category() == category:
+                category_list.append(item)
+                
+        return category_list
+
+    
+    def get_best_by_category(self,category):
+        """
+        Returns the item with the highest condition in the given category.
+        Returns None if no items in that category exist.
+        """
+        category_list = self.get_by_category(category)
+
+        if not category_list:
+            return None
+
+        best_item = category_list[0]
+
+        for item in category_list:
+            if item.condition > best_item.condition:
+                best_item = item
```

Comment:
If you were to use a built-in, which we'll allow in Unit 2, `max` would work nicely here:
```py
return max(category_list, key=lambda item: item.condition)